### ISSUE TITLE
Issues accessing `enum.nonmember` from another `nonmember`

### ISSUE THREAD

**Issue by max-muoto** [microsoft/pyright#7974]
Post 1.1.364 and [this issue](url) being resolved, this code snippet now passes Pyright:

```python
import enum

class MyEnum(enum.StrEnum):
    VAL_1 = "val_1"
    VAL_2 = "val_2"

    _items = enum.nonmember({VAL_1: 1, VAL_2: 2})

    def to_int(self) -> int:
        return self._items[self]
```

However, I'm running into issues with this exmaple:

```python
import enum


class MyEnum(enum.StrEnum):
    _FROM_TYPE = enum.nonmember(
        {
            0: "INT",
            1: "STRING",
        }
    )
    _TO_TYPE = enum.nonmember({v: k for k, v in _FROM_TYPE.items()})

```

Where I get `Cannot access attribute "items" for class "nonmember[dict[int, str]]`. You can resolve the warning by adding `.value` after `_FROM_TYPE`, but this would lead to a runtime error. Running into some issues with Playground on the latest version (indefinitely stalling), or else I would include links to these examples there.

**VS Code extension or command-line**
Pyright 1.1.364, CLI

**Comment by erictraut** [comment]
Modeling all behaviors of `EnumMeta` is probably not feasible in static analysis. It's ridiculously complex and rather dynamic in its behavior. At some point, I'm going to need to draw a line and say no. Pyright already goes way beyond other type checkers in its handling of `Enum` oddities. 

I'll look into whether I can support this pattern, but I may end up closing this as "won't fix", in which case you'll need to find a workaround in your code.

**Comment by max-muoto** [comment]
> Modeling all behaviors of `EnumMeta` is probably not feasible in static analysis. It's ridiculously complex and rather dynamic in its behavior. At some point, I'm going to need to draw a line and say no. Pyright already goes way beyond other type checkers in its handling of `Enum` oddities.
> 
> I'll look into whether I can support this pattern, but I may end up closing this as "won't fix", in which case you'll need to find a workaround in your code.

Sounds good, no worries if this can't be supported, I appreciate it nonetheless.

**Comment by erictraut** [comment]
There's not a straightforward way to implement this, so I'm going to close it as "won't fix". I'm willing to reconsider in the future if there's sufficient interest from other pyright users (e.g. upvotes on this issue).

You can work around this issue by moving the logic outside of the Enum class definition.

```python
_FROM_TYPE_VALUE = {0: "INT", 1: "STRING"}
_TO_TYPE_VALUE = {v: k for k, v in _FROM_TYPE_VALUE.items()}

class MyEnum(enum.StrEnum):
    _FROM_TYPE = enum.nonmember(_FROM_TYPE_VALUE)
    _TO_TYPE = enum.nonmember(_TO_TYPE_VALUE)
```

Another option is to use private names:

```python
    __FROM_TYPE_VALUE = {0: "INT", 1: "STRING"}
    _FROM_TYPE = enum.nonmember(__FROM_TYPE_VALUE)
    __TO_TYPE_VALUE = {v: k for k, v in __FROM_TYPE_VALUE.items()}
    _TO_TYPE = enum.nonmember(__TO_TYPE_VALUE)
```
